lib/clerk/sdk.rb

```diff
@@ -106,5 +110,40 @@ def sms_messages
     def users
       Resources::Users.new(self)
     end
+
+    # Returns the decoded JWT payload without verifying if the signature is
+    # valid.
+    #
+    # WARNING: This will not verify whether the signature is valid. You
+    # should not use this for untrusted messages! You most likely want to use
+    # verify_token.
+    def decode_token(token)
+      JWT.decode(token, nil, false).first
+    end
+
+    # Decode the JWT and verify it's valid (verify claims, signature etc.)
+    #
+    # JWKS are cached for JWKS_CACHE_LIFETIME seconds, in order to avoid
+    # unecessary roundtrips. In order to invalidate the cache, pass
+    # `force_refresh_jwks: true`.
+    def verify_token(token, force_refresh_jwks: false)
+      unverified_claims = decode_token(token)
+
+      jwk_loader = ->(options) do
+        @cached_jwks = nil if options[:invalidate] || force_refresh_jwks
+        @cached_jwks = nil if @last_fetched_at && Time.now.to_i - @last_fetched_at > JWKS_CACHE_LIFETIME
+
+        @cached_jwks ||= begin
+          @last_fetched_at = Time.now.to_i
+          conn = Faraday::Connection.new(unverified_claims["iss"])
+          response = conn.get("/.well-known/jwks.json")
+          raise "JWKs fetch error: #{response.status} #{response.body}" if !response.success?
+
+          JSON.parse(response.body, symbolize_names: true)
+        end
+      end
+
+      JWT.decode(token, nil, true, { algorithms: ['RS256'], jwks: jwk_loader})
```

Comment:
🔧 Make the algorithm an option that defaults to `RS256`. That way it will be more future proof

Reply:
done!